pkg/testers/node/node.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package node implements a node tester that implements e2e node testing following
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/e2e-node-tests.md#delete-instance-after-tests-run
+// https://github.com/kubernetes/kubernetes/blob/96be00df69390ed41b8ec22facc43bcbb9c88aae/build/root/Makefile#L206-L271
+// currently only support REMOTE=true
+package node
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/octago/sflags/gen/gpflag"
+	"k8s.io/klog"
+	"sigs.k8s.io/boskos/client"
+
+	"sigs.k8s.io/kubetest2/pkg/boskos"
+	"sigs.k8s.io/kubetest2/pkg/exec"
+)
+
+const (
+	target                 = "test-e2e-node"
+	gceProjectResourceType = "gce-project"
+)
+
+type Tester struct {
```

Comment:
how do we pass in `--image-config-file`? https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/runner/remote/run_remote.go#L55

Many tests pass in custom image file, e.g. https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/kubernetes/test-infra%24+file:%5Econfig/jobs+--image-config-file%3D&patternType=literal

Reply:
So this PR isn't the full migration, I anticipate we'll port over specific functionality as and when needed. Infact I'd prefer that so we don't just blindly port over all the flags.

I chose the subset of things used by `pull-kubernetes-node-e2e` as a start

Reply:
Note that power users can still set the corrsponding ENVs before invoking kubetest2 (until kubetest2 supports them as flags (if at all))

Reply:
Got it, makes sense, would be good clarify that in the PR description that goal here is just `pull-kubernetes-node-e2e` for now.

Reply:
Ack, updated PR description